compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -925,7 +925,7 @@ object SymDenotations {
     /** The top-level class containing this denotation,
      *  except for a toplevel module, where its module class is returned.
      */
-    final def topLevelClass(implicit ctx: Context): Symbol = {
+    def topLevelClass(implicit ctx: Context): Symbol = {
```

Comment:
Why not do `if (!exists) symbol` instead of adding an override? That's more consistent with other definitions in this class like `lexicallyEnclosingClass`

Reply:
Agreed, I'll change it.